### ISSUE TITLE
Fix false positives for multi-line selectors with comments in block-*-brace-*-*

### ISSUE THREAD

**Issue by ArmorDarks** [stylelint/stylelint#2761]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

A bug.

For some reasons some comments affecting listed below rules and forces them to report false errors.

> Which rule, if any, is this issue related to?

`block-opening-brace-newline-after`, `block-closing-brace-newline-before` and `block-opening-brace-space-after`

> What CSS is needed to reproduce this issue?

```scss
input, // 1
button { overflow: visible; }

summary, // a  test
section { display: block; }

input, /* 1 */
button { overflow: visible; }

summary, /* a  test */
section { display: block; }
```

> What stylelint configuration is needed to reproduce this issue?

```yaml
  block-opening-brace-newline-after: always-multi-line
  block-closing-brace-newline-before: always-multi-line
  block-opening-brace-space-after: always-single-line
```

> Which version of stylelint are you using?

 `8.0.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

CLI with `stylelint *.scss`"

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

Partially. Both Sass and regular CSS comments affects behavior.

> What did you expect to happen?

No warnings to be flagged.

> What actually happened (e.g. what warnings or errors you are getting)?

The following warnings were flagged:

```shell
source/_test.scss
  4:5   ×  Expected single space after "{" of a single-line block          block-opening-brace-space-after
  6:20  ×  Expected newline after "{" of a multi-line block                block-opening-brace-newline-after
  7:26  ×  Expected newline before "}" of a multi-line block               block-closing-brace-newline-before
  9:1   ×  Unexpected duplicate selector "input,  button", first used at   no-duplicate-selectors
           line 3
 10:2   ×  Expected single space after "{" of a single-line block          block-opening-brace-space-after
 12:1   ×  Unexpected duplicate selector "summary,  section", first used   no-duplicate-selectors
           at line 6
 12:20  ×  Expected newline after "{" of a multi-line block                block-opening-brace-newline-after
 13:26  ×  Expected newline before "}" of a multi-line block               block-closing-brace-newline-before
```

![sublime_text_2017-07-18_18-55-18](https://user-images.githubusercontent.com/4460311/28326907-ac1c58d6-6bea-11e7-9ef9-4ec13bcee4ef.png)

**Comment by jeddy3** [comment]
@ArmorDarks Thanks for the report.

I can replicate this. It is triggered whenever a comment exists anywhere, except on the last line, within a multi-line selector.

This will trigger it:

```css
a /* trigger */
b
c { prop: value; }
```

So will this:

```css
a 
b /* trigger */
c { prop: value; }
```

This won't:

```css
a
b
c /* no-trigger */ { prop: value; }
```

Labelling as `help wanted` and `bug`. All three rules make sure of the [`whitespaceChecker`](https://github.com/stylelint/stylelint/blob/master/lib/utils/whitespaceChecker.js) and [`rawNodeString`](https://github.com/stylelint/stylelint/blob/master/lib/utils/rawNodeString.js) utils. I suggest [beginning the investigation](https://github.com/stylelint/stylelint/blob/master/docs/developer-guide/rules.md#fixing-a-bug-in-an-existing-rule) there.

**Comment by hudochenkov** [comment]
[Demo with an issue](https://stylelint.io/demo#N4Igxg9gJgpiBcICWA7ADgVwC4BoAEA9AXgIwA6KARtlhCnsHhAG4wBOAZgDYQDu8eZkgDOSSlxgBuPAF8KFYRgC2SgIZsAnviJ5VePFhjCsCmGCxI6DPFBFouqjQPEQwAa2lyUFVJlyEAKlI8AIIKaixaekYWdm4+ASFRcSlZeRRFFXUtQN19Q2MQsIyzCytGW2F7R2ced08QHBBhDRQsVQAPBGawYWFG8DoOJABzbuB5ExA2DAlhMgQGSbIpl3cAWgg0GBRUEfXKNlUwGHWUGF4uVFPVDkM2BYEF1S5eR2F1pVmLdavzhZwy1WdTc6zAPFEKH2h2Op3Ol2uBxgHAgbBgjzwz1e70+3yQv2uAKBCzWoK2Oz2ByOJ3WVVh61u9wxWLeGg+kJGEgJ-xAky8MhAMiAA).

**Comment by jeddy3** [comment]
Closing [as stylistic rules are frozen](https://github.com/stylelint/stylelint/issues/5674). The community is welcome to migrate the rules [to a plugin](https://stylelint.io/developer-guide/plugins) and fix this bug.

**Comment by ArmorDarks** [comment]
It's a bug, not a proposal. Please, re-open the issue.
